### PR TITLE
Orca: Refactor Runner with ModelIO ABC

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

ModelIO takes charge of load and save stuffs.

All runners should inherit `ModelIO`` as the worker interface and implement the following APIs:
```python
    @abstractmethod
    def get_state_dict(self):
        """Returns the state of the runner."""
        pass

    @abstractmethod
    def load_state_dict(self, state):
        """Sets the state of the model."""
        pass

    @abstractmethod
    def save_checkpoint(self, filepath, save_weights_only=False):
        """Save checkpoint."""
        pass
```

https://github.com/intel-analytics/BigDL/pull/6408#issue-1432988407

### CODE REVIEW

File: python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py [hunk 1]
```diff
@@ -118,7 +115,7 @@ def __init__(self,
                             format='[%(asctime)s] %(levelname)-8s %(message)s',
                             datefmt='%Y-%m-%d %H:%M:%S'
                             )
-        self.logger = logging.getLogger(__name__)
+        BaseRunner.__init__(self, logging.getLogger(__name__))
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py:118]: super(TorchRunner).init?

**Review by hkvision — APPROVED** [review]

File: python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py (hunk 1 above)
Comment (leonardozcm) [reply python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py]: this is a typo, there is no need to init logger in baserunner(as logger is not used). removed now.

**Comment by jason-dai** [comment]
> ## Description
> ModelIO takes charge of load and save stuffs.
> 
> All runners should inherit `ModelIO`` as the worker interface and implement the following APIs:

Only PyTorch runners, yes?

**Comment by hkvision** [comment]
> > ## Description
> > ModelIO takes charge of load and save stuffs.
> > All runners should inherit `ModelIO`` as the worker interface and implement the following APIs:
> 
> Only PyTorch runners, yes?

Yes, only for PyTorch.

**Pull request merged**